registry/zookeeper/registry.go

```diff
@@ -418,19 +422,24 @@ func (r *zkRegistry) Subscribe(url *common.URL, notifyListener registry.NotifyLi
 			time.Sleep(time.Duration(RegistryConnDelay) * time.Second)
 			continue
 		}
-
+		n := 0
 		for {
+			n++
 			if serviceEvent, err := listener.Next(); err != nil {
 				logger.Warnf("Selector.watch() = error{%v}", perrors.WithStack(err))
-				listener.Close()
-				return
+				if n == 1 {
+					listener.Close()
```

Comment:
why n==1 listener.Close()?

Reply:
I'm thinking about how to manage waitGroup better